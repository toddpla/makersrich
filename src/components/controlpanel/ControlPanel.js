import React, { Component } from 'react';
import './control.css'
import NewsFeed from './NewsFeed'
import StatusBar from './StatusBar'

class ControlPanel extends Component {

  render() {
    return (
      <div id="control-panel">
        <StatusBar />
        <NewsFeed />
      </div>
    );
  }

}

export default ControlPanel;
