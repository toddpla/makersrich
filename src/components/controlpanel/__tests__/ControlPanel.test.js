import React from 'react'
import ControlPanel from '../ControlPanel'
import { shallow } from 'enzyme'

let wrapper

beforeEach(function() {
  wrapper = shallow(
    <ControlPanel />
  )
});

test('it renders ControlPanel', () => {
  expect(wrapper).toMatchSnapshot()
})
