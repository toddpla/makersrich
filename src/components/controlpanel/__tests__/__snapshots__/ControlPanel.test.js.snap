// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders ControlPanel 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="control-panel-button"
        >
          Leaderboard
        </button>,
        <Connect(StatusBar) />,
        <NewsFeed />,
        <Connect(MessageForm) />,
        <Connect(MessageFeed) />,
      ],
      "id": "control-panel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Leaderboard",
          "className": "control-panel-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Leaderboard",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleOffFocus": undefined,
          "handleOnFocus": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="control-panel-button"
          >
            Leaderboard
          </button>,
          <Connect(StatusBar) />,
          <NewsFeed />,
          <Connect(MessageForm) />,
          <Connect(MessageFeed) />,
        ],
        "id": "control-panel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Leaderboard",
            "className": "control-panel-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Leaderboard",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleOffFocus": undefined,
            "handleOnFocus": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
